src/main/scala/com/gossiperl/client/Messaging.scala

```diff
@@ -0,0 +1,73 @@
+package com.gossiperl.client
+
+import java.util
+
+import akka.actor.{Props, ActorLogging, Actor}
+import akka.util.Timeout
+import com.gossiperl.client.actors.ActorRegistry
+import com.gossiperl.client.exceptions.GossiperlClientException
+import com.gossiperl.client.serialization.{Serializer, DeserializeResultForward, DeserializeResultError, DeserializeResultOK}
+import com.gossiperl.client.thrift._
+import com.gossiperl.client.transport.{UdpTransportProtocol, UdpTransport}
+import scala.collection.JavaConverters._
+
+import scala.util.{Failure, Success}
+
+class Messaging(val configuration: OverlayConfiguration) extends ActorRegistry with ActorLogging {
+
+  import scala.concurrent.duration._
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  context.actorOf(Props( new UdpTransport(configuration) ), name = s"${configuration.overlayName}-transport")
+
+  log.debug(s"Messaging for overlay ${configuration.overlayName} is running.")
+
+  implicit val timeout = Timeout(1 seconds)
+
+  def receive = {
+    case GossiperlProxyProtocol.ShutdownRequest(p) =>
+      log.debug("Received shutdown request.")
+      p.success(self)
+      context.stop( self )
+    case UdpTransportProtocol.IncomingData(result) =>
+      result match {
+        case _:DeserializeResultOK =>
+          result.asInstanceOf[DeserializeResultOK].getDigestType match {
+            case Serializer.DIGEST =>
+              digestAck( result.asInstanceOf[DeserializeResultOK].getDigest.asInstanceOf[ Digest ] )
+            case Serializer.DIGEST_ACK =>
+              !:(s"${configuration.overlayName}-client-state", FSMProtocol.AckReceived)
```

Comment:
In akka, an actor is available either using a path (`/user/.../...`) or using (and passing) `ActorRef around`. In this actor system the assumption is that there's always a single instance of an actor with a specific name. The path is not important to me. I know that the name is something like:

 - `gossiperl_overlay_remote-client-state`
 - `gossiperl_overlay_remote-messaging`

Thus I assume that if I want to call an actor using this unique name, I do not have to pass the ref around, neither I have to know the path and use actorSelection (really cumbersome). `ActorRegistry` (which you commented on below) defines an operator `!:(actorName, message:Any, Option[Promise[ActorRef] = None)`, which uses an internal registry to reference actors by the name only. I am able to reference an actor by name and not having to care about the path - the path is not important to me. The fact that akka does this nesting depending on what actor starts what other actors bothers me a bit. Coming from Erlang - there's no such issue. Everything is on the same level, you an reference an `actor`, which in Erlang is just a process, by its name or pid (name is an atom, say `gossiperl_client_messaging`, pid is an address, something like: `<0.32.0>`. To send a message to such actor / process (they're the same thing) I can do either:

    Pid (read <0.32.0>) ! ...

or

    gossiperl_client_messaging ! ...

`!:(..., ...)` is the closest I can get to that functionality.